pytest/tests/test_sudo.py

```diff
@@ -0,0 +1,634 @@
+    provider.sudorule("test").add(user=g, host="ALL", command="/bin/ls")
+
+    client.sssd.common.sudo()
+    client.sssd.start()
+
+    # Until https://github.com/SSSD/sssd/issues/4483 is resolved
+    # Running 'id user-1' will resolve SIDs into group names
+    if isinstance(provider, AD):
+        client.tools.id("user-1")
+
+    assert client.auth.sudo.list("user-1", "Secret123", expected=["(root) /bin/ls"]), "Sudo list failed!"
+    assert client.auth.sudo.run("user-1", "Secret123", command="/bin/ls /root"), "Sudo command failed!"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.ticket(bz=1826272, gh=5119)
+@pytest.mark.topology(KnownTopology.BareAD)
+def test_sudo__user_is_nonposix_group(client: Client, provider: GenericADProvider):
+    """
+    :title: Non-POSIX groups can be set in sudoUser attribute
+    :setup:
+        1. Create user "user-1"
+        2. Create group "group-1" with "user-1" as a member
+        3. Create sudorule to allow "group-1" run "/bin/ls on all hosts
+        4. Enable SSSD sudo responder
+        5. Disable ldap_id_mapping
+        6. Start SSSD
+    :steps:
+        1. List sudo rules for "user-1"
+        2. Run "sudo /bin/ls" as "user-1"
+    :expectedresults:
+        1. User is able to run only /bin/ls
+        2. Command is successful
+    :customerscenario: False
+    """
+    u = provider.user("user-1").add(uid=10001, gid=10001)
+    g = provider.group("group-1").add().add_member(u)
+    provider.sudorule("test").add(user=g, host="ALL", command="/bin/ls")
+
+    client.sssd.common.sudo()
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.start()
+
+    assert client.auth.sudo.list("user-1", "Secret123", expected=["(root) /bin/ls"]), "Sudo list failed!"
+    assert client.auth.sudo.run("user-1", "Secret123", command="/bin/ls /root"), "Sudo command failed!"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.ticket(bz=1910131)
+@pytest.mark.topology(KnownTopology.BareAD)
+@pytest.mark.topology(KnownTopology.BareIPA)
+@pytest.mark.topology(KnownTopology.BareLDAP)
+def test_sudo__runasuser_shortname(client: Client, provider: GenericADProvider):
+    """
+    :title: sudoRunAsUser contains shortname
+    :setup:
+        1. Create user "user-1"
+        2. Create user "user-2"
+        3. Create sudorule to allow "user-1" run "/bin/ls on all hosts as "user-2" using shortname
+        4. Enable SSSD sudo responder
+        5. Start SSSD
+    :steps:
+        1. List sudo rules for "user-1"
+    :expectedresults:
+        1. User is able to run /bin/ls as "user-2"
+    :customerscenario: True
+    """
+    u1 = provider.user("user-1").add()
+    provider.user("user-2").add()
+    provider.sudorule("test").add(user=u1, host="ALL", command="/bin/ls", runasuser="user-2")
+
+    client.sssd.common.sudo()
+    client.sssd.start()
+
+    assert client.auth.sudo.list("user-1", "Secret123", expected=["(user-2) /bin/ls"]), "Sudo list failed!"
```

Comment:
Test if user is actually able to run command as expected

Reply:
What I meant here is :
`    assert client.auth.sudo.run("user-1", "Secret123", expected=["(user-2) /bin/ls"]), "Sudo list failed!"`
Along with `list`, execute the `run` as well.